### PR TITLE
support DATE can get datetime format string as arg

### PR DESCRIPTION
**Pull request by bbbearxyz**
**What type of PR is this?**

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

**Which issue(s) this PR fixes:**

issue #3143 

**What this PR does / why we need it:**

Not Available

**Special notes for your reviewer:**

Not Available

**Additional documentation (e.g. design docs, usage docs, etc.):**

Not Available

### CODE REVIEW

**Review by fengttt — COMMENTED** [review]
Do we really want date function return null instead of error?   I think we should error out.

File: pkg/vectorize/date/date.go [hunk 1]
```diff
@@ -22,3 +27,18 @@ func datetimeToDate(xs []types.Datetime, rs []types.Date) []types.Date {
 	}
 	return rs
 }
+
+func dateStringToDate(xs *types.Bytes, ns *nulls.Nulls, rs []types.Date) []types.Date {
+	for i := range xs.Lengths {
+		str := string(xs.Get(int64(i)))
+		d, e := types.ParseDatetime(str, 6)
+		if e != nil {
+			// set null
```
Comment (fengttt) [inline pkg/vectorize/date/date.go:36]: What about panic here?

Comment (bbbearxyz) [reply pkg/vectorize/date/date.go:36]: for example, date_add("xxxx", interval 1 day), the result will be null

Comment (bbbearxyz) [reply pkg/vectorize/date/date.go:36]: when the input args is "xxxx", we can't parse it so we will return null. For example, if the users use add_date(date()) from table. Table have many strings, but one of them is invalid to parse, I think return null will cost resources.

File: pkg/vectorize/date/date.go [hunk 2]
```diff
@@ -22,3 +27,18 @@ func datetimeToDate(xs []types.Datetime, rs []types.Date) []types.Date {
 	}
 	return rs
 }
+
+func dateStringToDate(xs *types.Bytes, ns *nulls.Nulls, rs []types.Date) []types.Date {
+	for i := range xs.Lengths {
+		str := string(xs.Get(int64(i)))
+		d, e := types.ParseDatetime(str, 6)
+		if e != nil {
```
Comment (nnsgmsone) [inline pkg/vectorize/date/date.go:35]: You may need to look at pgsql or sqlserver.

Comment (bbbearxyz) [reply pkg/vectorize/date/date.go:35]: yeah, I will change to return err, not set null when string is err

**Review by aressu1985 — APPROVED** [review]

**Review by fengttt — APPROVED** [review]

**Pull request merged**